src/lib/auth-utils.ts

```diff
@@ -204,12 +223,20 @@ export function parseAESCredentials(authorizationHeader: string, secret: string)
   }
 }
 
-export const expireReasons: string[] = ['JsonWebTokenError', 'TokenExpiredError'];
+export const expireReasons: string[] = ['JsonWebTokenError', 'TokenExpiredError', 'TokenRevokedError'];
 
-export function verifyJWTPayload(token: string, secret: string): RemoteUser {
+export async function verifyJWTPayload(token: string, secret: string, storage?: IStorageHandler): Promise<RemoteUser> {
   try {
-    const payload: RemoteUser = verifyPayload(token, secret);
-
+    const payload: RemoteUser&{jti?:string} = verifyPayload(token, secret);
+    // check for revoked tokens
+    if (storage && payload?.jti && payload?.name) {
+      const revoked = await isTokenRevoked(payload.name, payload.jti, storage);
+      if (revoked) {
+        const err = new Error("The JWT token is revoked by the user");
+        err.name = "TokenRevokedError";
```

Comment:
If the catch already returns `createAnonymousRemoteUser()` if there is an error, I supposed if the token is revoked you don't want to return an error, don't you?  At least this function job is return a payload, either if it is verified or anonymous.  This is related to my comment above of the name allowed as `void`. My opinion here is return `createAnonymousRemoteUser();`

Reply:
that makes sense, will adjust accordingly

Reply:
The background of that context, in a nutshell, I thought to remove AES, but then I realized we cannot, the comment is invalid most likely.

About the change, I disagree, the result would be the same, furthermore, I'd need to wrap with `try/catch` in every single place I use `verifyJWTPayload`. I find hacky the fact the `catch` throws an error, for me sounds unnatural and misleading from the develop point of view. Furthermore, `_handleJWTAPIMiddleware-> getMiddlewareCredentials-> verifyJWTPayload` uses this asynchronous function which at the same time is used by a `express` router, that's will lead to errors and is not officially supported https://github.com/expressjs/express/pull/2809